### ISSUE TITLE
Problem of log_sum_exp

### ISSUE THREAD

**Issue by MasakazuYposhimura** [amdegroot/ssd.pytorch#483]
I also encountered the same problem with #85 using my own dataset. 
Furthermore, in my case, inf didn't appear in the loss output because of the choices in hard negative mining although the inference result was very poor and it took much time to find where is wrong...

`x_max = x.data.max()`
`return torch.log(torch.sum(torch.exp(x-x_max), 1, keepdim=True)) + x_max`
This code solves the inf problem of exp but there are still the risks of -inf problem of log. 

So, I changed the code as follows and the program could predict properly.
`#x_max = x.data.max()`
`x_max, x_max_indices = torch.max(x, dim=1, keepdim=True)`
 `return torch.log(torch.sum(torch.exp(x-x_max), 1, keepdim=True)) + x_max`

I'm glad if this info is useful.

**Comment by zcy5417** [comment]
Thank you, you method worked!
and i think change the code in multibox_loss.py will also work:
`# loss_c = log_sum_exp(batch_conf) - batch_conf.gather(1, conf_t.view(-1, 1))`
`loss_c=F.cross_entropy(batch_conf,conf_t.view(-1))`

**Comment by calebx89** [comment]
> Thank you, you method worked! and i think change the code in multibox_loss.py will also work: `# loss_c = log_sum_exp(batch_conf) - batch_conf.gather(1, conf_t.view(-1, 1))` `loss_c=F.cross_entropy(batch_conf,conf_t.view(-1))`

take cross entropy will result in one number, what we want is the loss for each box.
